src/main/scala/esmeta/analyzer/AbsSemantics.scala

```diff
@@ -161,6 +161,41 @@ class AbsSemantics(
   def errors: Set[TypeError] =
     errorMap.map((_, innerMap) => innerMap.values.reduce(_ + _)).toSet
 
+  def unreachables: Set[Unreachable] = {
+    val funcs = (for {
+      func <- cfg.funcs
+      entry = func.entry
+    } yield func -> entry).toSet
+    var set = Set[Unreachable]()
+    val visited = MMap[Node, Boolean]()
+    val total = npMap.keys.map(_.node).toSet
+    def checkUnreachable(node: Node) = {
+      val states = npMap
+        .filter(_._1.node == node)
+        .values
+        .toSet
+      states.forall(_.isBottom)
+    }
+    def aux(node: Node)(using func: Func, cause: Option[Node] = None): Unit = {
+      if (visited.contains(node)) return
+      visited += node -> true
+      val unreachable = checkUnreachable(node)
+      if (unreachable) set += Unreachable(func, node, cause)
+      given Option[Node] =
+        if (unreachable && cause.isEmpty) Some(node) else cause
```

Comment:
Does it guarantee the analysis stops at only a single node per function?

Reply:
This is just for tracking `cause`(root cause node). Could you elaborate what `guarantee the analysis stop` means?